test/e2e/nodefeature/nodefeature.go

```diff
@@ -100,6 +100,10 @@ var (
 	// TODO: document the feature (owning SIG, when to use this feature for a test)
 	RuntimeHandler = framework.WithNodeFeature(framework.ValidNodeFeatures.Add("RuntimeHandler"))
 
+	// Added to test Swap Feature
+	// This label should be used when testing KEP-2400 (Node Swap Support)
+	Swap = framework.WithNodeFeature(framework.ValidNodeFeatures.Add("NodeSwap"))
```

Comment:
```suggestion
	Swap = framework.WithNodeFeature(framework.ValidNodeFeatures.Add("SwapConformance"))
```

I think.

Reply:
Not exactly.
The label you're pointing to here is for the NodeSwap feature gate. Instead I've added a `WithSwapConformance()` function to add the swap conformance label. Look at the [latest commit](https://github.com/kubernetes/kubernetes/pull/123557/commits/8a9a9617293551bddee0d4268fd7d0a71cedc1f5) for the implementation.

Reply:
Thats not true. This is just the label that we use for FOCUS.

Reply:
Right. That's what I meant. Let me rephrase.

We use `"NodeSwap"` at the upper `SIGDescribe`, meaning that this label appears for every swap test.
The new `"SwapConformance"` label introduced in this PR is applied only for the swap conformance tests.

This way:
- A job that focuses on NodeSwap runs all swap tests.
- A job that focuses on Serial will NOT run swap conformance.
- A job that focuses on SwapConformance will run swap conformance tests only.

Does this make sense? For more info look at the latest commit.

Reply:
So does "NodeSwap runs all swap tests" means all swap tests without SwapConformance?

Reply:
> So does "NodeSwap runs all swap tests" means all swap tests without SwapConformance?

No. Every swap test has the NodeSwap label. It makes sense, as if a job focuses only on NodeSwap it would be expected to run all swap related tests.

* All swap test has a NodeSwap label.
  * All swap node-conformance tests **also** have the NodeConformance label.
  * All swap swap-conformance tests **also** have the SwapConformance label.

Makes sense?

Please look at the code here which describes this:
https://github.com/iholder101/kubernetes/blob/635f936fcc3ffe902ef228defef4cfec367b2fbc/test/e2e_node/swap_test.go#L58-L85

Reply:
In short:
NodeConformance job would run only non-swap-conformance tests.
SwapConformance job would run only swap-conformance tests.

Reply:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-testing/3041-node-conformance-and-features could be a good read for you. 

NodeConformance is meant to be a test that can be run in all environments and not tied to any feature. SwapConformance seems to be meaningless and it is really a feature test of swap.

I'm going to tag @pohly and see what we think about `SwapConformance`.

Reply:
I just think that a test focuses on NodeSwap should run both node-conformance and non-node-conformance tests if they are related to the NodeSwap feature.

WDYT about replacing `SwapConformance` with something like `SerialSwap`?

In any case, if there's a strong push-back, I can add the NodeSwap label to serial tests only.

Reply:
I think having a swap enabled node run the node-conformance tests covers that case. So you could remove all swap related tests from the serial and non-serial swap jobs.

And in your "SwapConformance" jobs, you can just run all the tests that involve the swap feature. 

And then all the tests of the feature can be run on the serial node. You can test both NoSwap and LimitedSwap on serial because you can change the config.

I don't see a reason to add "SerialSwap". Serial is mostly just a implementation detail for the feature tests because we need to change from the default.

Reply:
@kannon92 
Alright, so let's summarize your suggestion to make sure we're on the same page here.

What you suggest is to remove `SwapConformance` from here:
https://github.com/iholder101/kubernetes/blob/635f936fcc3ffe902ef228defef4cfec367b2fbc/test/e2e_node/swap_test.go#L85

Then the swap-conformance jobs would focus on `NodeSwap` (i.e. `nodefeature.Swap`)?

This way node-conformance would run the non-stress swap tests
And swap-conformance would run ALL swap-related tests?

I'm fine with this approach

Reply:
Ideally, it would be nice to drop SwapConformance label entirely and just have all swap tests under NodeSwap. I think you titled the test-infra job for SwapConformance to target SwapConformance so we may need to change that first.

Reply:
@kannon92 Sure. Here's the PR: https://github.com/kubernetes/test-infra/pull/32276.

What are your thoughts on https://github.com/kubernetes/kubernetes/pull/123557#discussion_r1530313746?

Reply:
> BTW, I'm not sure what value we gain from running pull-kubernetes-node-swap-fedora-serial and pull-kubernetes-node-swap-ubuntu-serial, since they use NoSwap.

I think the goal was to make sure that enabling swap wouldn't break the conformance tests. I guess that since we defaulted from UnlimitedSwap to NoSwap we are only testing if swap will impact any of the tests.
> Isn't it (almost) similar to running the tests without swap enabled?
Don't we want to use LimitedSwap for these jobs?

We could but its not the default and changing a default in a non serial test is not really allowed. Changing a config requires a restart of kubelet..

We could change the serial jobs to use LimitedSwap but not sure what value that would have.

I could see us dropping those lanes eventually but I say we should keep them in case.

Reply:
> > BTW, I'm not sure what value we gain from running pull-kubernetes-node-swap-fedora-serial and pull-kubernetes-node-swap-ubuntu-serial, since they use NoSwap.
> 
> I think the goal was to make sure that enabling swap wouldn't break the conformance tests. I guess that since we defaulted from UnlimitedSwap to NoSwap we are only testing if swap will impact any of the tests.
> 
> > Isn't it (almost) similar to running the tests without swap enabled?
> > Don't we want to use LimitedSwap for these jobs?
> 
> We could but its not the default and changing a default in a non serial test is not really allowed. Changing a config requires a restart of kubelet..

Right. The lanes I've mentioned are serial.

> We could change the serial jobs to use LimitedSwap but not sure what value that would have.
> 
> I could see us dropping those lanes eventually but I say we should keep them in case.

I just think that at this point we have certain confidence in the fact that swap enabled on the node doesn't break serial tests. Running these tests with LimitedSwap would give raise our confidence to a much higher level since we'd know that not only a node with swap enabled doesn't break stuff, but actually using swap with LimitedSwap keeps the node stable.

Just a thought though, not a must.